libs/s25main/mapGenerator/HeadQuarters.h

```diff
@@ -47,22 +47,17 @@ namespace rttr { namespace mapGenerator {
     template<class T_Container>
     std::vector<MapPoint> FindHqPositions(const Map& map, const T_Container& area)
     {
-        std::vector<MapPoint> headQuarters;
-        for(const MapPoint& hq : map.headQuarters)
-        {
-            if(hq.isValid())
-            {
-                headQuarters.push_back(hq);
-            }
-        }
+        auto isHeadQuarter = [&map](const MapPoint& pt) {
+            return map.objectInfos[pt] == libsiedler2::OI_HeadquarterMask;
+        };
         auto isObstacle = [&map](MapPoint point) {
             return map.textureMap.Any(point, [](auto t) { return !t.Is(ETerrain::Buildable); });
         };
 
         // Quality of a map point as one player's HQ is a mix of:
         // 1. distance to other players' HQs (higher = better)
         // 2. distance to any other obstacle e.g. mountain & water (higher = better)
-        NodeMapBase<unsigned> potentialHeadQuarterQuality = DistancesTo(headQuarters, map.size);
+        NodeMapBase<unsigned> potentialHeadQuarterQuality = Distances(map.size, isHeadQuarter);
```

Comment:
Was this revert on purpose? I was only referring to the renaming hqPositions_->headQuarters